scheds/rust/scx_bpfland/src/bpf/main.bpf.c

```diff
@@ -8,23 +8,29 @@
 char _license[] SEC("license") = "GPL";
 
 /* Allow to use bpf_printk() only when @debug is set */
-#define dbg_msg(_fmt, ...) do {				\
-	if (debug)					\
-		bpf_printk(_fmt, ##__VA_ARGS__);	\
+#define dbg_msg(_fmt, ...) do { \
+	if (debug) \
+		bpf_printk(_fmt, ##__VA_ARGS__); \
 } while(0)
 
  /* Report additional debugging information */
 const volatile bool debug;
 
-/*
- * Priority DSQ used to dispatch interactive tasks.
- */
-static s32 prio_dsq_id;
+/* Global DSQ level iterator */
+enum {qidx_prio = 0, qidx_shared = 1};
+#define for_each_dsq_level(level) \
+	for(int level = 0; level < 2; level++)
+/* Cache layer iterator */
+enum {cidx_l3 = 0, cidx_l2 = 1, cidx_p = 2};
+int cla_disp_name(int index) {return 3 - index;}
+int codr_p_l2_l3 = 0xC6, codr_l2_l3_p = 0xE1, codr_l2_l3 = 0x31;
+#define for_each_cache_layer(layer, order) \
+	for (int layer = order; (layer & 3) != 3; layer >>= 2)
```

Comment:
I'm having a bit of a hard time to understand the logic here, especially the codr_p_* values... can you add a comment to explain the logic here, to make the code easier to read?

Reply:
Yes, sure.
I'll include the comments.
(Currently I'm having a hard time shooting the bug that's preventing the scx from starting, probably due my ignorance in BPF virtual machine's specification.)